packages/extras/src/lib/components/MeshLine/vertex.ts

```diff
@@ -20,62 +21,72 @@ export const vertexShader = `
     varying float vCounters;
 
-    vec2 fix( vec4 i, float aspect ) {
-        vec2 res = i.xy / i.w;
-        res.x *= aspect;
-        vCounters = counters;
-        return res;
+    vec2 intoScreen(vec4 i) {
+        return resolution * (0.5 * i.xy / i.w + 0.5);
     }
 
-    void main()	{
+    void main() {
+        float aspect = resolution.y / resolution.x;
 
-        float aspect = resolution.x / resolution.y;
+        mat4 m = projectionMatrix * modelViewMatrix;
 
-        vColor = vec4( color, opacity );
-        vUV = uv;
+        vec4 currentClip = m * vec4( position, 1.0 );
+        vec4 prevClip = m * vec4( previous, 1.0 );
+        vec4 nextClip = m * vec4( next, 1.0 );
 
-        mat4 m = projectionMatrix * modelViewMatrix;
-        vec4 finalPosition = m * vec4( position, 1.0 );
-        vec4 prevPos = m * vec4( previous, 1.0 );
-        vec4 nextPos = m * vec4( next, 1.0 );
+        vec4 currentNormed = currentClip / currentClip.w;
+        vec4 prevNormed = prevClip / prevClip.w;
+        vec4 nextNormed = nextClip / nextClip.w;
 
-        vec2 currentP = fix( finalPosition, aspect );
-        vec2 prevP = fix( prevPos, aspect );
-        vec2 nextP = fix( nextPos, aspect );
+        vec2 currentScreen = intoScreen(currentNormed);
+        vec2 prevScreen = intoScreen(prevNormed);
+        vec2 nextScreen = intoScreen(nextNormed);
 
-        float w = lineWidth * width;
+        float actualWidth = lineWidth * width;
 
         vec2 dir;
-        if( nextP == currentP ) dir = normalize( currentP - prevP );
-        else if( prevP == currentP ) dir = normalize( nextP - currentP );
-        else {
-            vec2 dir1 = normalize( currentP - prevP );
-            vec2 dir2 = normalize( nextP - currentP );
-            dir = normalize( dir1 + dir2 );
-
-            vec2 perp = vec2( -dir1.y, dir1.x );
-            vec2 miter = vec2( -dir.y, dir.x );
+        if(nextScreen == currentScreen) {
+            dir = normalize( currentScreen - prevScreen );
+        } else if(prevScreen == currentScreen) {
+            dir = normalize( nextScreen - currentScreen );
+        } else {
+            vec2 inDir = currentScreen - prevScreen;
+            vec2 outDir = nextScreen - currentScreen;
+            vec2 fullDir = nextScreen - prevScreen;
+
+            if(length(fullDir) > 0.0) {
+                dir = normalize(fullDir);
+            } else if(length(inDir) > 0.0){
+                dir = normalize(inDir);
+            } else {
+                dir = normalize(outDir);
+            }
         }
 
-        vec4 normal = vec4( -dir.y, dir.x, 0., 1. );
-        normal.xy *= .5 * w;
-        normal *= projectionMatrix;
+        vec2 normal = vec2(-dir.y, dir.x);
 
-        if( scaleDown > 0. ) {
-            float dist = length(currentP - prevP);
-            normal *= smoothstep(0.0, scaleDown, dist);
+        if(sizeAttenuation != 0.0) {
+            normal /= currentClip.w;
+						normal *= min(resolution.x, resolution.y) / 2.0;
```

Comment:
This factor could be adjusted to make the result similar to the previous implementation

Reply:
If this has no implications on for instance the possible range of attenuation, I'm fine with adjusting it like this 👍

Reply:
Yes, I have just had a play around with the examples and it looks like removing the `/ 2.0` at the end of line 70 will match the width to the original implementation. Currently the lines are half the width they should be.

Apart from that it all looks good to me!

The lines in the CursorLines example look very large because I'm using an orthographic camera so I should have set the attenuation to false. I'm happy to submit a PR to fix that.

Reply:
I have remove the `/ 2.0` scaling now.

Reply:
Thanks for your work on this @laszlokorte ! Happy for you to merge @grischaerbe 👍